packages/core/src/deal-tracker/spade-oracle-sync-tick.js

```diff
@@ -0,0 +1,212 @@
+// @ts-expect-error no types available
+import { ZSTDDecompress } from 'simple-zstd'
+import { Readable } from 'stream'
+// @ts-expect-error no types available
+import streamReadAll from 'stream-read-all'
+import { toString } from 'uint8arrays/to-string'
+import { encode, decode } from '@ipld/dag-json'
+import { RecordNotFoundErrorName } from '@web3-storage/filecoin-api/errors'
+import { parse as parseLink } from 'multiformats/link'
+
+/**
+ * @typedef {import('@web3-storage/filecoin-api/deal-tracker/api').DealStore} DealStore
+ * @typedef {import('./types').OracleContracts} OracleContracts
+ * @typedef {import('./types').SpadeOracle} SpadeOracle
+ * @typedef {import('../store/types').SpadeOracleStore} SpadeOracleStore
+ */
+
+/**
+ * On CRON tick, this function syncs deal store entries with the most up to date information stored
+ * in Spade's Oracle:
+ * - The previous oracle state known is fetched, as well as the latest state from Spade endpoint.
+ * - Once both states are in memory, they are compared and a diff is generated.
+ * - Diff is stored in deal store
+ * - Handled new state of oracle is stored for comparison in next tick.
+ * 
+ * @param {object} context
+ * @param {DealStore} context.dealStore
+ * @param {SpadeOracleStore} context.spadeOracleStore
+ * @param {URL} context.spadeOracleUrl
+ */
+export async function spadeOracleSyncTick ({
+  dealStore,
+  spadeOracleStore,
+  spadeOracleUrl
+}) {
+  // Get previous recorded spade oracle contracts
+  const getPreviousSpadeOracle = await getSpadeOracleState({
+    spadeOracleStore,
+    spadeOracleId: spadeOracleUrl.toString(),
+  })
+  if (getPreviousSpadeOracle.error && getPreviousSpadeOracle.error.name !== RecordNotFoundErrorName) {
+    return getPreviousSpadeOracle
+  }
+
+  // Get updated spade oracle contracts
+  const getUpdatedSpadeOracle = await getSpadeOracleCurrentState(spadeOracleUrl)
```

Comment:
Can we swap "spade oracle state" for "deal archive"? So this would be for example:

```suggestion
  const { ok: nextArchive, error } = await fetchDealArchive(spadeOracleUrl)
```